src/matchMaker/MatchMaker.py

```diff
@@ -1,58 +1,99 @@
+                                bare_metal[bare_metal_key] == request.requirements[bare_metal_key]:
+                    curr_req_score += self.calc_score(bare_metal_key)
 
-            # if total score of current request is bigger than the global value, update it
-            if curr_req_score > req_max_score['score']:
-                req_max_score = {'request': request, 'score': curr_req_score}
+                elif request.other_prop is not None and\
+                    bare_metal_key in request.other_prop and\
+                    bare_metal[bare_metal_key] == request.other_prop[bare_metal_key]:
+                        curr_req_score += self.calc_score(bare_metal_key)
 
-            # update the list with all requests
-            # Request.update_request_score(request, curr_req_score)
+                elif bare_metal_key in request.os and\
+                                bare_metal[bare_metal_key] == request[bare_metal_key]:
+                    curr_req_score += self.calc_score(bare_metal_key)
 
-        return req_max_score['request']
+            # compare by score
+            if curr_req_score > best_match_req['score']:
+                best_match_req = {'request': request, 'score': curr_req_score}
+            elif best_match_req['score'] != 0 and best_match_req['score'] == curr_req_score:
+                # compare by creation time
+                if request.creation_time > best_match_req['request'].creation_time:
+                    best_match_req = {'request': request, 'score': curr_req_score}
+        return best_match_req['request']
 
     @staticmethod
-    def calc_score(key):
-        # TODO read scores from a config file
+    def find_match_by_requirements(bare_metal, req_list):
+        """
+        finds requests in which all requirements are met
+        :param bare_metal:
+        :param req_list:
+        :return:
+        """
+        matched_req_by_requirements = []
+
+        for request in req_list:
+            request_match = True
+            if request.requirements is not None:
+                for requirements_key in request.requirements:
+                    if requirements_key not in bare_metal.keys():
+                        request_match = False
+                        break
+                    else:
+                        if request.requirements[requirements_key] != bare_metal[requirements_key]:
+                            request_match = False
+                            break
+            if request_match:
+                matched_req_by_requirements.append(request)
 
-        if key == "name":
-            score = 2
-        elif key == "url":
-            score = 4
+        return matched_req_by_requirements
+
+    # TODO decide what we prefer: there is another option, to use sections, and declare a default section,
+    #  so when the score is not declared, it will search it in defaule section and init score with 0
+    def calc_score(self, key):
+        score = 0
+
+        try:
```

Comment:
I think we should override get, so we won't need to know about SECTION and about what to except